### PR TITLE
Function Signatures and Build Integration

### LINKED ISSUE #5049: Crawl API for Function Signatures
For the last part of API crawling, we need to diff and crawl for function signatures.

### LINKED ISSUE #5050: Integrate API Crawling into Build Process
The build process should keep track of items deelted and added in the version changes, and upload them to a repository or S3 bucket in order to keep track of them.

### PR DESCRIPTION
**Pull request by brittainhard**
Finishes out API crawling feature, first parts of integration with the build process.
closes #5050 
closes #5049

### CODE REVIEW

**Comment by brittainhard** [comment]
@mattpap can you review this code? I'm working on the build script code now, but if you can take a look at the code so far I would appreciate it.

File: bokeh/util/api_crawler.py [hunk 1]
```diff
@@ -1,8 +1,15 @@
-import ast, os, copy
+from __future__ import print_function
+import ast, os, sys
 
 __all__ = ["api_crawler", "differ"]
 
 
+if sys.version_info > (3, 0):
```
Comment (mattpap) [inline bokeh/util/api_crawler.py]: `if six.PY3`

**Comment by brittainhard** [comment]
@mattpap can you take a look at this when you have some time?

**Comment by brittainhard** [comment]
@mattpap If you can find some time to do a code review it would be appreciated. I'd like to get this merged for 0.12.3.

**Comment by bryevdv** [comment]
@brittainhard there are some minor flake8 issues to fix:
```
>       assert proc.returncode == 0, "Flake8 issues:\n%s" % out.decode("utf-8")
E       AssertionError: Flake8 issues:
E         bokeh/util/tests/test_api_crawler.py:5:1: F811 redefinition of unused 'pytest' from line 3
E         bokeh/util/tests/test_api_crawler.py:358:161: E501 line too long (184 > 160 characters)
E         
E       assert 1 == 0
E        +  where 1 = <subprocess.Popen object at 0x7f0352f39510>.returncode

```

**Comment by brittainhard** [comment]
@mattpap can you review this code?

**Comment by mattpap** [comment]
That's a lot of code. I have some comments, but maybe would could improve this incrementally later.

**Comment by brittainhard** [comment]
@mattpap in that case should we merge it?

**Comment by bryevdv** [comment]
We are going to have a `0.12.4` very soon it looks like (to get in some features a client needs). Let's put this in `0.12.4` and give @mattpap time for a real review.

**Comment by bryevdv** [comment]
Not sure when we will be able to get back to this. I am closing this PR to clean up the list, but keeping the branch to base later work off of.

**Pull request closed without merging**